### PR TITLE
DF Documentation 🔖 

### PR DESCRIPTION
**Pull request by 9and3**
## description

This is a PR introducing the first documentation framework for DF. It's not complete I just throw in the main structure and we will have to split the work to have a nice fully documentation for DF 😃 .
To know how to document I  wrote a small guide in `Further resources` >> `Developer documentation` >> `Documentation`. 

I liked a lot Jax ( thanks @DamienGilliard for the ref!) and I tried to replicate for us, have a look at the screenshot plis. Basically it is split in two:

* a) `Getting Started`: top user (top hand) aimed to be a quickstart basically and that's our gh user, 
* b) `Further resources`: for developers that need extra resources. 

Have a look and tell me what you think about the general structure and bare-bone content. *We can maybe merge it and then split the work to do for the tutorial documentation later*.

This is what it looks like:

image

image

image

>[!WARNING]
>I cannot show you the documentation since it has to be merged on main first before to be deployed  but you can switch do this branch, upload the conda and do:
>```console
>invoke documentize
>start ./_build/index.html
>```
>this will build and show locally the  static website.

For all the features have a look here..

## main contribs/modifications 
* *unecessary docstrings*: I got rid of the docstrings in the `Run()` function for all the `codee.py` in the components. This function is not needed anymore since a double of the metadata (that's what we use for generating documentation)
* *component doc*: I introduced a sphinx extension to automate the documentation of the components
* *py source code doc*: typical py source is documented in sphinx mode. I forsee a refactoring of the `pybind` module that is  not fully-automatically documented (marked on #64 ).
* *pre-commits*: added precommit and linter `Ruff` and `pymy` to help us check python code sanity before committing. Before committing and in an action you will see something like this:
image
* *better documentation*: since we are going open source, it seemed good to have a better documentation on how to develop and contribute. It isn't finished but a big chunck is there and I will add the rest in a near future.

## additional info
The building of the documentation and deployment is grouped in one git workflow in doc-build.yml. Only when merged to main the documentation will be deployed to the GithubPage, in all other scenarios it will only test the building of the doc.

image

### CODE REVIEW

**Comment by 9and3** [comment]
I am missing a couple of documents to migrate from CONTRIBUTING.md but I am already opening it for review.

**Comment by eleniv3d** [comment]
Looks very slick to me! 👓

**Comment by 9and3** [comment]
Ok I am merging this one, thanks @9and3 for the feedback!

File: doc/sphinx_ghcomponent_parser/metadata_parser.py
Comment (DamienGilliard) [inline doc/sphinx_ghcomponent_parser/metadata_parser.py:1]: nice !

Comment (DamienGilliard) [reply doc/sphinx_ghcomponent_parser/metadata_parser.py:1]: No other remark, looks sleek indeed

**Pull request merged**